### DISCUSSION TITLE
Reboot device

### DISCUSSION THREAD

**Discussion by StephenD52** [jasonacox/tinytuya#535] [Q&A]
Hello, this is my first post.  I've just started using tinytuya to monitor 4 identical Circuit Breakers that use Tuya Smart.

Does anyone know if it is possible to remotely reboot (rather than factory reset) a device and, if so, how?

I have 4 (new) circuit breakers all of which work fine with the Tuya Smart app and 3 of them work fine with tinytuya but I can't connect locally to the 4th device with tinytuya.  A tinytuya scan finds the device but, after using the wizard, a scan fails to pull data from the device.  The response is:
{
            "id": "xxxxxxxxxxxxxxxxxxxxxxxx",
            "ip": "192.168.0.58",
            "active": 2,
            "encrypt": true,
            "productKey": "pqcmqh1ufgmalru0",
            "token": true,
            "wf_cfg": true,
            "name": "Breaker 1",
            "key": "^ac(10rgDAv41y29",
            "mac": "fc:3c:d7:10:a5:3c",
            "ability": 0,
            "dev_type": "default",
            "err": "No response",
            "origin": "broadcast",
            "ver": "3.4"
        },
The output from the other 3 breakers includes the dps values.

(I can pull data from all devices using tinytuya.Cloud.) 

My first thought was that the Local Key was wrong, or somehow was not correctly interpreted by tinytuya, but I also think the app might also have difficulty connecting locally:  If I use Tuya Smart on the local network it cannot perform a "Check Device Network" on this device whereas it can on the others.  (The app says "The device and mobile need to use the same WiFi" so I assume this test is done locally whereas other settings use the cloud.)

I'm working on this remotely and the devices are unattended and 2000 miles away so I cannot perform a power cycle reboot, which might kick it into life, or an unpair/pair operation to create a new local key, which might also work.

I'm thinking if there is a way to reboot the device, perhaps using Tuya Developer, I could try the first option without a 4 hour flight each way!  All I can find is a factory reset. 

Many thanks.

**Comment by jasonacox** [comment]
Hi @StephenD52 - I really wish there was a way to reboot these devices! I have yet to find it. I have devices that get into a bad state where I lose local control even if cloud control works. It is clearly a bug in the firmware that makes the devices brittle like that. Most users would never know since cloud based app control still works. That's the only reason I can figure as to why they never fix these.

If only we could reboot them... One possible path would be to trick the device into doing a firmware update (even if it is current) resulting in a firmware flash and reboot, but I couldn't find a path to do that. The Tuya IoT docs do talk about resetting, but it would unbind and I'm not sure how that would work for you if you are remote: https://developer.tuya.com/en/docs/iot-device-dev/TuyaOS-iot_abi_device_reset?id=Kc67srci7m1jk - And I haven't even tried that myself since power cycling works (yes, I know, not helpful if you are dealing with embedded circuit breakers). 

I would love it if there was a list of Tuya devices that rated the local API reliability.  But the problem with that is that the Tuya makers change so frequently that it is hard to create an evergreen list that would survive beyond a year.

If anyone has any tricks to get them to reboot remote, please let us know. 🙏

**Comment by uzlonewolf** [comment]
I've looked and haven't been able to find a way to reboot devices remotely.  The firmware images I've reverse engineered have no way of rebooting via a local connection, and while it looks like you should be able to reboot devices via a Cloud command I have not been able to find a public API endpoint to actually do it.

**Comment by StephenD52** [comment]
Thanks for the input.  I definitely don't want to unbind the device when I am not physically present because I will loose long distance control via the app, which works OK.  I guess I'll have to wait until my next visit to try to work out a "cure".

I'm really wanting to continuously monitor a parameter from one of the breakers and, luckily, it's not the offending one, so lack of tinytuya contact with just the bad one it is not mission critical.

Thanks for your help.

**Reply by StephenD52** [reply]
Just an update to my issue, I recently made the 4000 mile round trip to the device's location and while there did a reboot (power cycle).  The device started to work with tinytuya without having to unbind/bind.
